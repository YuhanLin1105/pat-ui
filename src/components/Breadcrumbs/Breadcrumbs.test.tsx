import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Bread from './Breadcrumbs';
import BreadOption from './BreadcrumbsOption';

describe('Breadcrumb', () => {
    it(' should match snapshot', () => {
        const { asFragment } = render(<Bread>default bread</Bread>)
        expect(asFragment()).toMatchSnapshot();
    })

    it(' should render default Breadcrumbs', () => {
        const breadProps = {
            className: 'test',
        }
        const wrapper = render(<Bread { ...breadProps }>default bread</Bread>);
        const element = wrapper.queryByRole('bread') as HTMLElement;
        // const element = wrapper.queryByText('⮞ default bread') as HTMLElement;
        expect(element).toBeInTheDocument()
        expect(element.tagName).toBe('UL')
        expect(element).toHaveClass('bread')
        console.log(element.tagName)
    })

    it(' should render primary Breadcrumbs', () => {
        const breadProps = {
            className: 'test',
        }
        const wrapper = render(<Bread {...breadProps}>primary sm bread</Bread>);
        // const element = wrapper.queryByText('⮞ primary sm bread') as HTMLElement;
        // expect(element).toBeInTheDocument()
        // expect(element.tagName).toBe('UL')
        // expect(element).toHaveClass('bread test')
    })

    it(' should render BreadOption', () => {
        const breadOptionProps = {
            value: 'val',
            onClick: jest.fn(),
            className: 'test',
        }

        const wrapper = render(<BreadOption {...breadOptionProps}>breadoption default</BreadOption>);
        const element = wrapper.queryByText('breadoption default') as HTMLElement;
        expect(element).toBeInTheDocument()
        expect(element.tagName).toBe('LI')
        expect(element).toHaveClass('bread__option test')
        expect(breadOptionProps.onClick).toHaveBeenCalledTimes(0)
        fireEvent.click(element)
        expect(breadOptionProps.onClick).toHaveBeenCalledTimes(1)
    })

    it(' should render divider slash', () => {
        const wrapper = render(<BreadOption>About</BreadOption>);
        const element = wrapper.queryByText('About') as HTMLElement;
        expect(element).toBeInTheDocument();
    })
})