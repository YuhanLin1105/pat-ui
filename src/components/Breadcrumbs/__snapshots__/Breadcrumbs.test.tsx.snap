// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb  should match snapshot 1`] = `
<DocumentFragment>
  <ul
    class="bread bread-default bread-arrow"
    role="bread"
  >
    <li
      class="home"
    >
      Home 
    </li>
     default bread
  </ul>
</DocumentFragment>
`;
